app/apphandlers/embeddedAppHandler.go

```diff
@@ -172,20 +182,22 @@ func (embedded *EmbeddedApp) StartRudderCore(ctx context.Context, options *app.O
 		modeProvider = state.NewStaticProvider(servermode.DegradedMode)
 	}
 
-	proc := processor.New(ctx, &options.ClearDB, gwDBForProcessor, routerDB, batchRouterDB, errDB, multitenantStats, reportingI)
+	proc := processor.New(ctx, &options.ClearDB, gwDBForProcessor, routerDB, batchRouterDB, errDB, multitenantStats, reportingI, transientSources)
```

Comment:
I used DI for many years and I'm not a big fan to be honest, I'll try to explain why since now it's been a few years without and I'm not looking back.

I think it's yet another thing that we put on top and that one has to learn before being able to contribute. It bloates the code, adds lines to debug and lines to the stack and it makes the binary heavier. Also, and of course it is subjective, I personally prefer to initialize things manually and they are usually not that complex or boring to setup if you follow the SOLID principles (emphasis on the `S` of SOLID, if you're unit really has a single responsibility I'm pretty sure it'll have only a small set of things that you need to inject). By doing that we'd only have very few structs like the processor that would need lots of deps and DI wouldn't bring enough to the table to justify a new framework in the project.

Reply:
Yes, you always have to give something in order to take something else. I have used DI for many years as well, although on a different language, and it something that I am missing here, where we keep on solving the dependency injection problem in all sorts of other different ways :)

I am not expecting us to use DI, it is hard to retrofit such a design choice anyway. Just thought that this is a good example to argue in favour of DI in general :)